io/plugins/eu.esdihumboldt.hale.io.gml.test/src/data/shiporder/shiporder.xml

```diff
@@ -14,11 +14,16 @@ xsi:schemaLocation="http://www.example.com shiporder.xsd">
     <title>Empire Burlesque</title>
     <note>Special Edition</note>
     <quantity>1</quantity>
-    <price>10.90</price>
+    <price>0E-11</price>
```

Comment:
The root issue is that this notation is actually not valid and that hale studio should not create such a notation.

Fine for me to add a test for reading it, nice that we can deal with it, but it should definitely be mentioned that this is not valid according to the schema, so this can't result in any misunderstandings.

What we need to test in addition though, is writing data that would have resulted in an invalid representation before your changes, and verify that the notation in the XML is valid (and not what we get when we read it with hale).

So as before my suggestion would be to (in addition) create tests similar to in `StreamGmlWriter2Test`, because they provide a good template for writing test data given an XML schema. You can create an XML schema that is suited for your specific test cases for it and create minimal instance examples as required. As mentioned, `XmlSlurper` would be an easy way to access the content of the created XML file to verify the string representation in the file.